core/src/main/java/com/linecorp/armeria/common/ThreadLocalRequestContextStorage.java

```diff
@@ -46,7 +46,7 @@ public void pop(RequestContext current, @Nullable RequestContext toRestore) {
         requireNonNull(current, "current");
         final InternalThreadLocalMap map = InternalThreadLocalMap.get();
         final RequestContext contextInThreadLocal = context.get(map);
-        if (current != contextInThreadLocal) {
+        if (current.unwrapAll() != contextInThreadLocal.unwrapAll()) {
```

Comment:
Let's also check if `contextInThreadLocal` is null or not just in case `pop` is called without `push`.